arch/cortex-m/samv71-hal/src/hal.rs

```diff
@@ -110,49 +106,116 @@ impl SystemHal for Hal {
     type Duration = crate::time::TimerDurationU64<{ Hal::TIMER_FREQ }>;
     type Error = HalError;
 
-    fn configure_hardware(&mut self, config: SystemHardwareConfig) -> Result<(), HalError> {
-        // SAFETY: This is safe, because this is a single-core system,
-        // and no other references to system peripherals should exist.
-        let peripherals = unsafe { self.system_peripherals.as_mut_ref() };
+    /// Initialize global HAL instance using PAC peripherals.
+    ///
+    /// Calling this function begins HAL initialization process. This process must be finished
+    /// by calling [`SystemHal::configure_hardware`]. Until then, no other HAL functions should
+    /// be called, as they will most likely fail.
+    ///
+    /// This function executes in critical section, as it modifies HAL_SYSTEM_PERIPHERALS.
+    ///
+    /// # Safety
+    /// This function is safe to call only once.
+    /// Subsequent calls will return an error, indicating that HAL instance has already been created.
+    ///
+    /// # Return
+    /// `()` on success, [`HalError::HalAlreadyCreated`] if called more than once.
+    fn create() -> Result<(), HalError> {
```

Comment:
Is this necessary to have `fn create` and `fn configure_hardware` separate? If one has to be called after another, and `create` doesn't return anything it seems unnecessary.

Reply:
You're right, i've refactored this into a single function